applications/OptimizationApplication/python_scripts/responses/standardized_objective.py

```diff
@@ -0,0 +1,65 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_info import OptimizationInfo
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.communicators.response_function_communicator import ResponseFunctionCommunicator
+
+class StandardizedObjective:
+    """Standardized objective response function
+
+    This class creates instances to standardize any response function (make it a minimization problem).
+    Supported objective types:
+        "minimization",
+        "maximization"
+
+    """
+    def __init__(self, parameters: Kratos.Parameters, optimization_info: OptimizationInfo):
+        default_parameters = Kratos.Parameters("""{
+            "response_name": "",
+            "type"         : "",
+            "scaling"      : 1.0
+        }""")
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        self.__objective_type = parameters["type"].GetString()
+        if self.__objective_type == "minimization":
+            self.__scaling = parameters["scaling"].GetDouble()
+        elif self.__objective_type == "maximization":
+            self.__scaling = -parameters["scaling"].GetDouble()
+        else:
+            raise RuntimeError(f"Requesting unsupported type {self.__objective_type} for objective response function. Supported types are: \n\tminimization\n\tmaximization")
+
+        self.__communicator = ResponseFunctionCommunicator(parameters["response_name"].GetString(), optimization_info)
+        self.__initial_response_value = None
+
+    def GetResponseFunctionName(self) -> str:
+        return self.__communicator.GetName()
+
+    def GetInitialValue(self) -> float:
+        if self.__initial_response_value is None:
+            self.__initial_response_value = self.__communicator.GetScaledValue()
+        return self.__initial_response_value
+
+    def GetResponseType(self) -> str:
+        return self.__objective_type
+
+    def GetStandardizedValue(self, step_index: int = 0) -> float:
```

Comment:
here the same, since you are scaling I would call the method GetScaledValue, the name does not reflect what it does unless one check the implementation

Reply:
I would keep the name as it is, because this makes sense from the perspective of `Algorithms`. This does scaling as well as the negation for `maximization` objectives. I added more elaborate documentation for the method.